Initial commit for Branch-Split job

AdamBrousseau opened the pull request:
- Pipeline file to branch OpenJ9 and OpenJ9-OMR
- Fail if branch already exists
- Fail if split point does not exist
- Inputs: new branch name and SHAs for OpenJ9 and OMR

[skip ci]

Signed-off-by: Adam Brousseau <adam.brousseau88@gmail.com>

AdamBrousseau commented:
Jenkins Job: https://ci.eclipse.org/openj9/job/Branch-Split
Example of a successful split: https://ci.eclipse.org/openj9/job/Branch-Split/13/
Example of a duplicate branch name: https://ci.eclipse.org/openj9/job/Branch-Split/14/
Example of trying to split from a sha that doesn't exist: https://ci.eclipse.org/openj9/job/Branch-Split/15/

AdamBrousseau commented:
I will amend the commit with the Eclipse repos once we are satisfied with it.

pshipton commented on buildenv/jenkins/jobs/infrastructure/branchSplit:
```diff
@@ -0,0 +1,115 @@
+
+HTTP = 'https://'
+OMR_REPO = 'github.com/AdamBrousseau/openj9-omr.git'
+OPENJ9_REPO = 'github.com/AdamBrousseau/openj9.git'
```
These don't look right.

AdamBrousseau replied:
Right, Once we are happy with the file I will amend with `eclipse`. Keeping it mine for now in case I need to test any changes.

pshipton replied:
Guess I should have read your comments first. :)

pshipton commented:
Note the branch name format change for the example https://github.com/eclipse/openj9/pull/1199.
The job could also take a tag name, as the release process calls for the branch to be immediately tagged.

AdamBrousseau commented:
>The job could also take a tag name, as the release process calls for the branch to be immediately tagged

I had thought of doing that. But we are also doing a tag job. So I figured it was better to not include the "auto-tag" in the branch job. Up for discussion though.

pshipton commented:
It easier to run one job than two. I was thinking this job could take an "initial tag" parameter as well, which could be left blank.

DanHeidinga commented on buildenv/jenkins/jobs/infrastructure/branchSplit:
```diff
@@ -0,0 +1,121 @@
+            echo "Check if SPLIT POINT: '${SPLIT_SHA}'' exists"
+            sh "git rev-parse --verify '$SPLIT_SHA'"    
+        } catch(e) {
+            MESSAGE = "Split Point: '${SPLIT_SHA}'' does not exixt in ${REPO_NAME}, stopping build."
```
nitpit: `exixt` -> `exist`

AdamBrousseau replied:
fixed

DanHeidinga commented:
> It easier to run one job than two. I was thinking this job could take an "initial tag" parameter as well, which could be left blank.

My vote is to build the branch and tag jobs separately as reusable components.  We can always add a job that invokes them one after the other to get Pete's desired behaviour

DanHeidinga commented on buildenv/jenkins/jobs/infrastructure/branchSplit:
```diff
@@ -0,0 +1,121 @@
+OPENJ9_REPO = 'github.com/AdamBrousseau/openj9.git'
+
+DEFAULT_SPLIT_OPENJ9 = 'master'
+DEFAULT_SPLIT_OMR = 'openj9'
```
Can you clarify the meaning of `DEFAULT_SPLIT_`?  It might just be a naming issue but I'm not 100% sure what this means.

AdamBrousseau replied:
If a sha is not provided, we default to splitting from tip of master/openj9

AdamBrousseau commented:
updated default split points to include `refs/heads/<branch>`
example of supplying no shas: https://ci.eclipse.org/openj9/job/Branch-Split/19/

AdamBrousseau commented:
if there are no other concerns, I will update the repo values and unmark WIP.

AdamBrousseau commented:
Ready for merging if no other concerns

AdamBrousseau commented:
Simplified script. Removed 2 checks for existing branch and valid branch_point. Git branch will fail if either is invalid and stderr will indicate what failed.
